src/Catel.MVVM/Catel.MVVM.Shared/Collections/SuspensionContext.cs

```diff
@@ -15,11 +16,22 @@ namespace Catel.Collections
     /// <typeparam name="T">Type of the elements contained by the suspending collection.</typeparam>
     internal class SuspensionContext<T>
     {
+        /// <summary>
+        /// The events generator registry.
+        /// </summary>
+        private static Dictionary<SuspensionMode, Func<SuspensionContext<T>, ICollection<NotifyRangedCollectionChangedEventArgs>>> _eventGeneratorsRegistry;
```

Comment:
Ok, if you want to only instantiate this if it is required, we have to use Lazy with the auto-property. Sry.